erxupdate/ntopnghosts.sh

```diff
@@ -0,0 +1,429 @@
+	if [[ "$DEBUG" != "" ]]; then 
+		trap - DEBUG
+		trap
+	fi
+	while IFS= read -r line; do
+		HOSTNAME=`echo "$line" | cut -d ' ' -f 1`
+		MAC=`echo "$line" | cut -d ' ' -f 2`
+		#echo "$HOSTNAME $MAC"
+		if [[ "$HOSTNAME" != "" ]]; then
+			if [[ "`echo -e "$HOSTLOG" |grep -m 1 "^$HOSTNAME $MAC"`" == "" ]]; then
+				# we didn't find a matching device that we've already processed
+				HOSTLOG="$HOSTLOG$CRLF$line"
+			fi
+		fi
+	done < /tmp/hostnameregister 
+	if [[ "$DEBUG" != "" ]]; then 
+		set -x
+		trap 'read -p "run: $BASH_COMMAND"' DEBUG
+	fi
+	#echo -e "$HOSTLOG"
+	echo -e "$HOSTLOG" | sort -f > /tmp/hostnameregister
+	SAVEPERIOD=1
+	# when did we last update the long term storage with our current last seen hosts list?
+	if [[ -f "/tmp/hostsupdate" ]]; then
+		LASTSAVEDHOSTS=`cat /tmp/hostsupdate`
+		SINCESAVED=`dateDiff -d "now" "$LASTSAVEDHOSTS"`
+	else 
+		# this is the first time we've run it
+		LASTSAVEDHOSTS=
+		SINCESAVED=9999999
+	fi
+	if [[ "$SINCESAVED" -gt "$SAVEPERIOD" ]]; then 
+		# now we need to write an update of all the hosts we can see today in dhcp
+		if [[ ! -f "/etc/nprobe/hostnameregister" ]]; then
+			touch /etc/nprobe/hostnameregister
+		fi
+		LONGTERMFILESIZE=`wc -c /etc/nprobe/hostnameregister|cut -d ' ' -f1`
+		TEMPORARYFILESIZE=`wc -c /tmp/hostnameregister|cut -d ' ' -f1`
+		if [[ "$LONGTERMFILESIZE" != "$TEMPORARYFILESIZE" ]]; then
+			cp /tmp/hostnameregister /etc/nprobe/hostnameregister
+			echo `date -I` > /tmp/hostsupdate
+			echo Updated the hostnameregister in long term storage
+		fi
+		# if we are centrally managing the hostnamesubstitute file then we may need to update it to long term storage too
+		if [[ "$NAMESUBSTITUTEURL" != "" ]]; then
+			if [[ ! -f "/etc/nprobe/hostnamesubstitute" ]]; then
+				touch /etc/nprobe/hostnamesubstitute
+			fi
+			LONGTERMFILESIZE=`wc -c /etc/nprobe/hostnamesubstitute|cut -d ' ' -f1`
+			TEMPORARYFILESIZE=`wc -c /tmp/hostnamesubstitute|cut -d ' ' -f1`
+			if [[ "$LONGTERMFILESIZE" != "$TEMPORARYFILESIZE" ]]; then
+				cp /tmp/hostnamesubstitute /etc/nprobe/hostnamesubstitute
+				echo Updated the hostnamesubstitute in long term storage
+			fi
+		fi
+	fi
+fi
+# we need to check if the script itself needs to be updated
+if [[ "$UPDATEURL" != "" ]]; then
+	UPDATESIZE=`curl -sI $UPDATEURL --location --silent -H 'Accept-Encoding: gzip,deflate' |grep -i content-length |cut -d ' ' -f2`
+	UPDATESIZE=(${UPDATESIZE:0:-1})
+	#curl -sI $UPDATEURL --location --silent --write-out 'size_download=%{size_download}\n'
+	if [[ -f "/etc/nprobe/ntopnghosts.sh" ]]; then
+		FILESIZE=`wc -c /etc/nprobe/ntopnghosts.sh|cut -d ' ' -f1`
+	else
+		FILESIZE=0
+	fi
+	#echo "Centrally managed script size: $UPDATESIZE"
+	#echo "Local script size: $FILESIZE"
+	if [[ "$UPDATESIZE" != "0" ]]; then
+		if [[ "$FILESIZE" != "$UPDATESIZE" ]]; then
+			echo "Downloading updated script"
+			curl -s $UPDATEURL -o "/tmp/ntopnghosts.sh"
```

Comment:
This can open up to a series of security issues as discussed above.

Reply:
switched to --ssl to require ssl upgrade